src/charm.py

```diff
@@ -327,6 +341,31 @@ def healthy(self) -> bool:
 
         return True
 
+    def update_client_data(self) -> None:
+        """Writes necessary relation data to all related client applications."""
+        if not self.unit.is_leader() or not self.healthy:
+            return
+
+        for client in self.state.clients:
+            if not client.password:
+                logger.debug(
+                    f"Skipping update of {client.app.name}, user has not yet been added..."
+                )
+                continue
+
+            client.update(
+                {
+                    "endpoints": client.bootstrap_server,
+                    "zookeeper-uris": client.zookeeper_uris,
+                    "consumer-group-prefix": client.consumer_group_prefix,
```

Comment:
**issue** I find it very weird having the schema of the relation in the charm and not in core. Ideally the name of the databag fields should only be centralized in `core`. The other components should only use APIs

Reply:
Let's address it later consistently in Kafka and ZooKeeper